We need to adapt the aufs patches to our kernel
These changes are needed because of the following patch:

commit 6637ecd306a94a03dd5b8e4e8d3f260d9877c5b0
Author: Al Viro <viro@zeniv.linux.org.uk>
Date:   Sun Oct 26 19:19:16 2014 -0400

    move d_rcu from overlapping d_child to overlapping d_alias

    commit 946e51f2bf37f1656916eb75bd0742ba33983c28 upstream.

    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
    Cc: Ben Hutchings <ben@decadent.org.uk>
    [hujianyang: Backported to 3.10 refer to the work of Ben Hutchings in 3.2:
     - Apply name changes in all the different places we use d_alias and d_child
     - Move the WARN_ON() in __d_free() to d_free() as we don't have dentry_free()]
    Signed-off-by: hujianyang <hujianyang@huawei.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Upstream-Status: Inappropriate [configuration specific]

Signed-off-by: Florin Sarbu <florin@resin.io>
---
 fs/aufs/dcsub.c   | 4 ++--
 fs/aufs/hnotify.c | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/fs/aufs/dcsub.c b/fs/aufs/dcsub.c
index 1746e79..c2e69fe 100644
--- a/fs/aufs/dcsub.c
+++ b/fs/aufs/dcsub.c
@@ -133,7 +133,7 @@ resume:
 	while (next != &this_parent->d_subdirs) {
 		struct list_head *tmp = next;
 		struct dentry *dentry = list_entry(tmp, struct dentry,
-						   d_u.d_child);
+						   d_child);
 
 		next = tmp->next;
 		spin_lock_nested(&dentry->d_lock, DENTRY_D_LOCK_NESTED);
@@ -169,7 +169,7 @@ resume:
 		this_parent = tmp;
 		spin_lock(&this_parent->d_lock);
 		rcu_read_unlock();
-		next = child->d_u.d_child.next;
+		next = child->d_child.next;
 		goto resume;
 	}
 
diff --git a/fs/aufs/hnotify.c b/fs/aufs/hnotify.c
index 6b5b85e..3dc01f9 100644
--- a/fs/aufs/hnotify.c
+++ b/fs/aufs/hnotify.c
@@ -378,7 +378,7 @@ static struct dentry *lookup_wlock_by_name(char *name, unsigned int nlen,
 
 	dentry = NULL;
 	spin_lock(&parent->d_lock);
-	list_for_each_entry(d, &parent->d_subdirs, d_u.d_child) {
+	list_for_each_entry(d, &parent->d_subdirs, d_child) {
 		/* AuDbg("%.*s\n", AuDLNPair(d)); */
 		spin_lock_nested(&d->d_lock, DENTRY_D_LOCK_NESTED);
 		dname = &d->d_name;
-- 
2.7.4

